PIMbench/cpp-brightness/baselines/CPU/brightness.cpp

```diff
@@ -0,0 +1,182 @@
+/**
+ * @file brightness.cpp
+ * @brief Template for a Host Application Source File.
+ *
+ */
+
+#include <stdio.h>
+#include <strings.h>
+#include <string.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <assert.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <ctype.h>
+#include <bits/stdc++.h>
+#include <omp.h>
+
+#include "../../../utilBaselines.h"
+
+#define MINCOLORVALUE 0 // Sets the max value that any color channel can be in a given pixel
+#define MAXCOLORVALUE 255 // Sets the max value that any color channel can be in a given pixel 
+
+// Params ---------------------------------------------------------------------
+typedef struct Params
+{
+  std::string inputFile;
+  bool shouldVerify;
+  int brightnessCoefficient;
+} Params;
+
+void usage()
+{
+  fprintf(stderr,
+          "\nUsage:  ./brightness.out [options]"
+          "\n"
+          "\n    -i    24-bit .bmp input file (default=uses 'sample1.bmp' from '/cpp-histogram/histogram_datafiles' directory)"
+          "\n    -v    t = verifies PIM output with host output. (default=false)"
+          "\n    -b    brightness coefficient value (default=20)"
+          "\n");
+}
+
+struct Params getInputParams(int argc, char **argv)
+{
+  struct Params p;
+  p.inputFile = "../../../cpp-histogram/histogram_datafiles/sample1.bmp";
+  p.shouldVerify = false;
+  p.brightnessCoefficient = 20;
+
+  int opt;
+  while ((opt = getopt(argc, argv, "h:i:v:b:")) >= 0)
+  {
+    switch (opt)
+    {
+    case 'h':
+      usage();
+      exit(0);
+      break;
+    case 'i':
+      p.inputFile = optarg;
+      break;
+    case 'v':
+      p.shouldVerify = (*optarg == 't') ? true : false;
+      break;
+    case 'b':
+      p.brightnessCoefficient = strtoull(optarg, NULL, 0);
+      break;
+    default:
+      fprintf(stderr, "\nUnrecognized option!\n");
+      usage();
+      exit(0);
+    }
+  }
+  return p;
+}
+
+uint8_t truncate(uint8_t pixelValue, int brightnessCoefficient)
+{
+  return static_cast<uint8_t> (std::min(MAXCOLORVALUE, std::max(MINCOLORVALUE, pixelValue + brightnessCoefficient)));
+}
+
+void brightness(std::vector<uint8_t> &resultData, uint64_t imgDataBytes, int brightnessCoefficient)
+{ 
+  #pragma omp parallel for
+  for (uint64_t i = 0; i < imgDataBytes; ++i) {
+    resultData[i] = truncate(resultData[i], brightnessCoefficient);
+  }
+}
+
+int main(int argc, char *argv[]) 
+{      
+  struct Params params = getInputParams(argc, argv);
+  std::string fn = params.inputFile;
+  std::cout << "Running brightness on CPU for input file : '" << fn << "'" << std::endl;
+
+  int fd;
+  uint64_t imgDataBytes, tempImgDataOffset;
+  struct stat finfo;
+  char *fdata;
+  unsigned short *dataPos;
+  int imgDataOffsetPosition;
+
+  // Start data parsing
+  if (!fn.substr(fn.find_last_of(".") + 1).compare("bmp") == 0)
```

Comment:
Please use C++ style and handle the case of no dot in file name:
if (fn.substr(fn.find_last_of(".")) != ".bmp")